stdlib/internal/SwiftSyntax/Trivia.swift

```diff
@@ -0,0 +1,279 @@
+  let pieces: [TriviaPiece]
+
+  /// Creates Trivia with the provided underlying pieces.
+  public init(pieces: [TriviaPiece]) {
+    self.pieces = pieces
+  }
+  
+  public init(from decoder: Decoder) throws {
+    var container = try decoder.unkeyedContainer()
+    var pieces = [TriviaPiece]()
+    while let piece = try container.decodeIfPresent(TriviaPiece.self) {
+      pieces.append(piece)
+    }
+    self.pieces = pieces
+  }
+  
+  public func encode(to encoder: Encoder) throws {
+    var container = encoder.unkeyedContainer()
+    for piece in pieces {
+      try container.encode(piece)
+    }
+  }
+
+  /// Creates an iterator that can iterate over the pieces of this Trivia.
+  public func makeIterator() -> AnyIterator<TriviaPiece> {
+    var index = pieces.startIndex
+    return AnyIterator {
+      guard index < self.pieces.endIndex else { return nil }
+      defer { index += 1 }
+      return self.pieces[index]
+    }
+  }
+
+  /// Creates Trivia with no pieces.
+  public static var zero: Trivia {
+    return Trivia(pieces: [])
+  }
+
+  /// Creates a new `Trivia` by appending the provided `TriviaPiece` to the end.
+  public func appending(_ piece: TriviaPiece) -> Trivia {
+    var copy = pieces
+    copy.append(piece)
+    return Trivia(pieces: copy)
+  }
+
+  /// Return a piece of trivia for some number of space characters in a row.
+  public static func spaces(_ count: Int) -> Trivia {
+    return [.spaces(count)]
+  }
+
+  /// Return a piece of trivia for some number of tab characters in a row.
+  public static func tabs(_ count: Int) -> Trivia {
+    return [.tabs(count)]
+  }
+
+  /// A vertical tab '\v' character.
+  public static func verticalTabs(_ count: Int) -> Trivia {
+    return [.verticalTabs(count)]
+  }
+
+  /// A form-feed '\f' character.
+  public static func formfeeds(_ count: Int) -> Trivia {
+    return [.formfeeds(count)]
+  }
+
+  /// Return a piece of trivia for some number of newline characters
+  /// in a row.
+  public static func newlines(_ count: Int) -> Trivia {
+    return [.newlines(count)]
+  }
+  
+  /// Return a piece of trivia for some number of backtick '`' characters
+  /// in a row.
+  public static func backticks(_ count: Int) -> Trivia {
+    return [.backticks(count)]
+  }
+
+  /// Return a piece of trivia for a single line of ('//') developer comment.
+  public static func lineComment(_ text: String) -> Trivia {
+    return [.lineComment(text)]
+  }
+
+  /// Return a piece of trivia for a block comment ('/* ... */')
+  public static func blockComment(_ text: String) -> Trivia {
+    return [.blockComment(text)]
+  }
+
+  /// Return a piece of trivia for a single line of ('///') doc comment.
+  public static func docLineComment(_ text: String) -> Trivia {
+    return [.docLineComment(text)]
+  }
+
+  /// Return a piece of trivia for a documentation block comment ('/** ... */')
+  public static func docBlockComment(_ text: String) -> Trivia {
+    return [.docBlockComment(text)]
+  }
```

Comment:
Similarly, I disagree with gyb'ing this.  It's much easier to read as plain source code and the benefit is very small.